app/controllers/dashboard/form/publish_controller.rb

```diff
@@ -30,6 +30,7 @@ def update
           # WorkVersion#set_thumbnail_selection may be unreliable if the Shrine::ThumbnailJob is delayed
           @resource.set_thumbnail_selection
           @resource.indexing_source = Proc.new { nil }
+          @resource.set_publisher_as_scholarsphere if deposit_pathway.instrument?
```

Comment:
This logic could also potentially go in the model...a `before_save` or even as part of the act of publishing.